test/cli_volume_test.go

```diff
@@ -305,49 +299,64 @@ func (suite *PouchVolumeSuite) TestVolumeListOptions(c *check.C) {
 	}
 
 	volumeName := "volume_" + funcname
-	volumeName1 := "volume_" + funcname + "_1"
+	volumeName1 := volumeName + "_1"
 	command.PouchRun("volume", "create", "--name", volumeName1, "-o", "opt.size=1g").Assert(c, icmd.Success)
 	defer command.PouchRun("volume", "rm", volumeName1)
 
-	volumeName2 := "volume_" + funcname + "_2"
+	volumeName2 := volumeName + "_2"
 	command.PouchRun("volume", "create", "--name", volumeName2, "-o", "opt.size=2g").Assert(c, icmd.Success)
 	defer command.PouchRun("volume", "rm", volumeName2)
 
-	volumeName3 := "volume_" + funcname + "_3"
+	volumeName3 := volumeName + "_3"
 	command.PouchRun("volume", "create", "--name", volumeName3, "-o", "opt.size=3g").Assert(c, icmd.Success)
 	defer command.PouchRun("volume", "rm", volumeName3)
 
 	// test --size and --mountpoint options
 	ret := command.PouchRun("volume", "list", "--size", "--mountpoint")
 	ret.Assert(c, icmd.Success)
 
-	for _, line := range strings.Split(ret.Stdout(), "\n") {
-		if strings.Contains(line, volumeName) {
-			if !strings.Contains(line, "local") ||
-				!strings.Contains(line, DefaultVolumeMountPath) {
-				c.Errorf("list result have no driver or name or size or mountpoint, line: %s", line)
-				break
-			}
+	lines := volumesToKV(ret.Stdout())
+	for _, line := range lines {
+		c.Assert(line[0], check.Equals, "local")
+		if !strings.Contains(line[3], DefaultVolumeMountPath) {
```

Comment:
I think when you were refactoring this part, you thought of add more checking rules which is described in https://github.com/alibaba/pouch/issues/2497 😄 @ZYecho

Reply:
yep, `check.Contains` is really helpful.